### PR TITLE
MMT-3624: umm_c's Spatial Information Bounding Rectangle boxes are missing i-icons, page is being rolled when using mouse roll to increase/decrease number in these boxes.

### PR DESCRIPTION
**Pull request by mandyparson**
# Overview

### What is the feature?

umm_c's Spatial Information --> Bounding Rectangle boxes need  i-icons
arrows need to be removed in lat/long widget

### What is the Solution?

Added CustomWidgetWrapper to each bounding rectangle coordinate and adjusted CustomWidgetWrapper to get it styled correctly. Added some scss rules to BoundingRectangle scss file

### What areas of the application does this impact?

CustomWidgetWrapper
BoundingRectangle

# Testing

### Reproduction steps

- **Environment for testing: local
- **Collection to test with: make a collection draft

1. Under Collection draft, go to Spatial information > Horizontal > Bounding Rectangle
2. Check that all cardinal directions have an information i that works when you hover over it
3. Check that only the Bounding Rectangle Boxes have the spinner buttons removed. 

### Attachments

Before
Screenshot 2024-07-08 at 2 46 45 PM

After
Screenshot 2024-07-08 at 2 46 06 PM

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/components/BoundingRectangleField/BoundingRectangleField.jsx [hunk 1]
```diff
@@ -15,9 +16,17 @@ import './BoundingRectangleField.scss'
...
  */
 const BoundingRectangleField = ({
+  name,
   formData,
-  onChange
+  onChange,
+  schema
 }) => {
+  const { properties } = schema
+  const { EastBoundingCoordinate: EBoundingCoordinate } = properties
```
Comment (macrouch) [inline static/src/js/components/BoundingRectangleField/BoundingRectangleField.jsx:25]: Variable names with upper cased first letters typically indicate they are the name of a class, so this should be renamed to `eastBoundingCoordinate`.

Avoid shortening or abbreviating variable names, the more descriptive they are, the easy they are to read and maintain.

You can destructure both coordinates from `properties` in one call


```
const { 
  EastBoundingCoordinate: eastBoundingCoordinate,
  NorthBoundingCoordinate: northBoundingCoordinate
} = properties
```

File: static/src/js/components/BoundingRectangleField/__tests__/BoundingRectangleField.test.jsx [hunk 2]
```diff
@@ -32,17 +42,19 @@ describe('BoundingRectangleField', () => {
     test('renders the fields', () => {
       setup()
 
-      const west = screen.getByLabelText('West')
-      expect(west).toHaveValue(-179)
+      const spinbuttons = screen.getAllByRole('spinbutton')
```
Comment (macrouch) [inline static/src/js/components/BoundingRectangleField/__tests__/BoundingRectangleField.test.jsx:45]: Why was this changed? This is a less descriptive way to find each field. Having to look at array indexes (`spinbuttons[0]`) is harder to maintain and more prone to errors than finding a field by label text.

File: static/src/js/components/CustomWidgetWrapper/CustomWidgetWrapper.jsx [hunk 3]
```diff
@@ -33,7 +34,8 @@ const CustomWidgetWrapper = ({
   maxLength,
   required,
   scrollRef,
-  title
+  title,
+  centered
```
Comment (macrouch) [inline static/src/js/components/CustomWidgetWrapper/CustomWidgetWrapper.jsx:38]: Sort keys alphabetically to keep readability high

File: static/src/js/components/BoundingRectangleField/BoundingRectangleField.jsx [hunk 4]
```diff
@@ -15,9 +16,17 @@ import './BoundingRectangleField.scss'
  * @param {BoundingRectangleField} props
  */
 const BoundingRectangleField = ({
+  name,
```
Comment (macrouch) [inline static/src/js/components/BoundingRectangleField/BoundingRectangleField.jsx:19]: Sort keys alphabetically to keep readability high

File: static/src/js/components/CustomWidgetWrapper/CustomWidgetWrapper.jsx [hunk 5]
```diff
@@ -45,13 +47,15 @@ const CustomWidgetWrapper = ({
     setShowHelp(false)
   }
 
+  const predeterminedJustification = centered ? 'justify-content-center' : 'justify-content-between'
```
Comment (macrouch) [inline static/src/js/components/CustomWidgetWrapper/CustomWidgetWrapper.jsx:50]: The `classnames` library should be used when conditionally adding classes to elements. Search the project for `classNames(` to find a good example

File: static/src/js/components/BoundingRectangleField/BoundingRectangleField.jsx [hunk 6]
```diff
@@ -3,6 +3,7 @@ import PropTypes from 'prop-types'
 import Button from 'react-bootstrap/Button'
 
 import './BoundingRectangleField.scss'
+import CustomWidgetWrapper from '../CustomWidgetWrapper/CustomWidgetWrapper'
```
Comment (macrouch) [inline static/src/js/components/BoundingRectangleField/BoundingRectangleField.jsx:6]: The style import should be the last import in the file, `import './BoundingRectangleField.scss'`. We do this for readability and consistency across the repo

**Comment by macrouch** [comment]
Can you add `"stylelint.vscode-stylelint"` to the array of recommendations in `.vscode/extensions.json`, that extension should aid in getting the stylelint step to pass

File: static/src/js/components/BoundingRectangleField/BoundingRectangleField.jsx (hunk 1 above)
Comment (mandyparson) [reply static/src/js/components/BoundingRectangleField/BoundingRectangleField.jsx:25]: I went with e instead of east because eastBoudningCoordinate is already being used elsewhere, but yes I can change it otherwise.

File: static/src/js/components/BoundingRectangleField/__tests__/BoundingRectangleField.test.jsx (hunk 2 above)
Comment (mandyparson) [reply static/src/js/components/BoundingRectangleField/__tests__/BoundingRectangleField.test.jsx:45]: because the other ones weren't working and the spin buttons don't have names. They are harder to target without test-ids. Any other suggestions? I tried screen.getByText and screen.getByRole('spinbutton', ( id: 'north')} but neither did the trick. I'll do some more digging on my end and see if I can find something better.

File: static/src/js/components/BoundingRectangleField/BoundingRectangleField.jsx (hunk 1 above)
Comment (macrouch) [reply static/src/js/components/BoundingRectangleField/BoundingRectangleField.jsx:25]: Ah ok, yeah I think eslint will complain about that. I'd probably add `Properties` to the end, or something like that. `eastCoundingCoordinateProperties`

File: static/src/js/components/BoundingRectangleField/__tests__/BoundingRectangleField.test.jsx (hunk 2 above)
Comment (macrouch) [reply static/src/js/components/BoundingRectangleField/__tests__/BoundingRectangleField.test.jsx:45]: There is still a label that is supposed to be attached to the input fields, but the `id` you are passing in to `CustomWidgetWrapper` broke that link. 

Previously the label was setup like this 
```
<label htmlFor="north-coordinate">North</label>
```

But you are passing in `name` to the `id` prop, and `CustomWidgetWrapper` uses `htmlFor={id}` in the label.

Aside from the tests, you should be able to see is broken by clicking on the label text in your browser. That should focus the correct input field, which is why we want our test to use `getByLabelText`, to ensure that our tests properly interact with the application in a way a user would.

File: static/src/js/components/CustomWidgetWrapper/CustomWidgetWrapper.jsx [hunk 7]
```diff
@@ -140,7 +144,8 @@ CustomWidgetWrapper.defaultProps = {
   scrollRef: null,
   maxLength: null,
   required: null,
-  title: null
+  title: null,
```
Comment (dmistry1) [inline static/src/js/components/CustomWidgetWrapper/CustomWidgetWrapper.jsx:147]: Sort these alphabetically

File: static/src/js/components/CustomWidgetWrapper/CustomWidgetWrapper.jsx [hunk 8]
```diff
@@ -151,7 +156,8 @@ CustomWidgetWrapper.propTypes = {
   maxLength: PropTypes.number,
   required: PropTypes.bool,
   scrollRef: PropTypes.shape({}),
-  title: PropTypes.string
+  title: PropTypes.string,
```
Comment (dmistry1) [inline static/src/js/components/CustomWidgetWrapper/CustomWidgetWrapper.jsx:159]: Sort these alphabetically

File: static/src/js/components/BoundingRectangleField/BoundingRectangleField.scss [hunk 9]
```diff
@@ -21,5 +21,10 @@
     width: 200px;  
     text-align: center;
   }
-}
 
+  input::-webkit-outer-spin-button,
```
Comment (dmistry1) [inline static/src/js/components/BoundingRectangleField/BoundingRectangleField.scss:25]: This change should be done globally so all of the number input field do not have the scroll wheel behavior

Comment (mandyparson) [reply static/src/js/components/BoundingRectangleField/BoundingRectangleField.scss:25]: The AC only has it being removed in this particular area >> https://bugs.earthdata.nasa.gov/browse/MMT-3624. I can't remember, did we verbally agree that we don't want them anywhere? If so, I'll update the AC

**Review by dmistry1 — APPROVED** [review]

**Review by macrouch — APPROVED** [review]
